### ISSUE TITLE
how to set ManufacturerData?

### ISSUE THREAD

**Issue by fvisztra** [nettlep/gobbledegook#7]
hi,

I love gobbledegook, but can't find how to set ManufacturerData (to advertise) :(

cheers

**Comment by fvisztra** [comment]
hello!

could anyone help me?

**Comment by rmartens** [comment]
Did you try replacing "Acme Inc." with the name of your company in line 273 in Server.cpp?

[Source line 273](https://github.com/nettlep/gobbledegook/blob/a7a0a74aa8d2a817ea9250dd3b93edf19ad0549e/src/Server.cpp#L273)

**Comment by fvisztra** [comment]
hi rmartens,

thank you for your answer but I'd like to advertise manufacturer data (as it is advertising tx power and short name ) and not modify characteristic

**Comment by rmartens** [comment]
Oh yeah, As far as I can tell this library doesn't provide a way to add all advertising data. The library would need to implement a org.bluez.LEAdvertisement1 and then register it with bluez. RegisterAdvertisement. From: [bluez advertising-api](https://git.kernel.org/pub/scm/bluetooth/bluez.git/tree/doc/advertising-api.txt)
Note: This library (gobbledegook) uses direct hci calls for some calls to bluetooth. This might be a bit confusing to bluetoothd which is suppose to handle all the kernel calls in newer versions of bluez. I'd think it would be better to replace all the hci calls with bluez dbus calls and let bluetoothd handle talking to the kernel module.

**Comment by fvisztra** [comment]
thank you!
 to be honest I am a newbie in ble developing, this is why I am using your solution!
is it big task to register  "org.bluez.LEAdvertisement1"?
if it is not so big could you send me a sample? I don't know where I should start...
I just would like to advertise just a number or a string...

thank you

**Comment by rmartens** [comment]
btw - this is not my solution, I'm an observer like you. I have played around with the project and tested a few changes. 
You will need to use and understand making dbus calls to the bluez process. Basically get the org.bluez.LEAdvertisingManager1 and then register your own org.bluez.LEAdvertisement1.
To get you started I forked this repo and the first [commit gets org.bluez.LEAdvertisingManager1](https://github.com/rmartens/gobbledegook/commit/7c4e21dc35cdbccf9ebc625a7d198ee82514145e)
